commands/shell/tty_unix.go

```diff
@@ -0,0 +1,52 @@
+// +build linux darwin dragonfly freebsd netbsd openbsd
+
+package shell
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/burke/ttyutils"
+)
+
+// SetupRawTerminal will set terminal to RAW mode and return function to
+// restore normal terminal.
+func SetupRawTerminal(setSize func(cols, row uint16) error) func() {
+	// Switch terminal to raw mode
+	state, err := ttyutils.MakeTerminalRaw(os.Stdout.Fd())
+	if err != nil {
+		// If we can't do it just continue as we were
+		return func() {}
+	}
+
+	// Get TTY size
+	size, err := ttyutils.Winsize(os.Stdout)
+	if err == nil && setSize != nil {
+		setSize(size.Columns, size.Lines)
+	}
+
+	// Handle SIGWINCH signals (window resize signals from graphical terminals)
+	sigWinch := make(chan os.Signal, 1)
+	signal.Notify(sigWinch, syscall.SIGWINCH)
+	go func() {
+		for range sigWinch {
+			size, err := ttyutils.Winsize(os.Stdout)
+			if err == nil && setSize != nil {
+				setSize(size.Columns, size.Lines)
+			}
```

Comment:
I don't think it has practical consequences... and it shouldn't leave the window in an incorrect size..
Because the `sigWinch` channel has capacity 1, so if a resize event happens, while processing the previous resize event... Then once the first resize event has been processed, the while-loop will get another message (the second resize event) from the `sigWinch` channel..

I didn't realize this until you pointed it out and forced me to consider it. But by having a capacity of one it doesn't matter how many resize events happen while we're processing the first resize event. It's true that only one even will be added to the channel, but since the message in the channel doesn't carry the window size it should all be okay...

Hmm... Perhaps the golang signal handling stuff isn't so crazy afterall.. I originally thought it was a bit weird.